### PR TITLE
fs: Add fchstat and chstat callback into mountpt_operations

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
and implement all status related change function. the individual
file system change will provide in other upcoming patchset.

## Impact
Support the new API(chmod, chown, utimes...)

## Testing
Please ignore these case warning:
```
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:850:12: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:1022:16: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:1422:13: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:1820:11: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:1821:11: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:1832:8: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/fs/nfs/nfs_vfsops.c:1853:42: error: Mixed case identifier found
```

### CODE REVIEW

File: fs/vfs/fs_fchstat.c [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+ * Name: fchstat
+ ****************************************************************************/
+
+static int fchstat(int fd, FAR struct stat *buf, int flags)
+{
+  FAR struct file *filep;
+  int ret;
+
+  /* First, get the file structure.  Note that on failure,
+   * fs_getfilep() will return the errno.
```
Comment (acassis) [inline fs/vfs/fs_fchstat.c:49]: It is not returning the "errno", but setting the errno and returning ERROR (-1). Please find this comment.

File: fs/vfs/fs_fchstat.c [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    {
+      goto errout;
+    }
+
+  /* Perform the fchstat operation */
+
+  ret = file_fchstat(filep, buf, flags);
+
+  /* Check if the fchstat operation was successful */
+
```
Comment (acassis) [inline fs/vfs/fs_fchstat.c:63]: Remove remove this comment (it is not necessary here), just:
  ret = file_fchstat(filep, buf, flags);
  if (ret >= 0)
    {
      /* Successfully fchstat'ed the file */

      return OK;
    }

File: fs/vfs/fs_fchstat.c [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+  DEBUGASSERT(filep != NULL);
+
+  /* Get the inode from the file structure */
+
+  inode = filep->f_inode;
+  DEBUGASSERT(inode != NULL);
+
+  /* Adjust and check buf and flags */
+
+  if ((flags & CH_STAT_MODE) && (buf->st_mode & ~07777))
```
Comment (acassis) [inline fs/vfs/fs_fchstat.c:117]: I think this test could always return true: (buf->st_mode & ~07777)
st_mode is of type mode_t that is 16-bit, but 07777 octal value is only 12-bit long, so if the upper 4 bits from st_mode are 0 the resulting value will be different of zero. Did you check it?

Comment (xiaoxiang781216) [reply fs/vfs/fs_fchstat.c:117]: the up 4bits is used to repesent file type and can not be modified by caller, this check ensure it.

File: fs/vfs/fs_fchstat.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_fchstat.c:49]: I can not understand your meaning, do you want me change the comment? but the comment is right becasue fs_getfilep do not set errno in failure path.

File: fs/vfs/fs_fchstat.c (hunk 2 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_fchstat.c:63]: Done

**Comment by xiaoxiang781216** [comment]
CI fail due to libstdc++ call utimensat wrongly:
https://github.com/llvm/llvm-project/blob/main/libcxx/src/filesystem/filesystem_common.h#L30-L36
Let's wait @Donny9 ' xxxat patch to fix this issue first.

**Comment by xiaoxiang781216** [comment]
> CI fail due to libstdc++ call utimensat wrongly:
> https://github.com/llvm/llvm-project/blob/main/libcxx/src/filesystem/filesystem_common.h#L30-L36
> Let's wait @Donny9 ' xxxat patch to fix this issue first.

Since utimensat need more time to support, I add a new patch to temporarily undefine UTIME_OMIT in C++ to avoid libc++ compiler error. The CI should pass and the PR is ready to merge, @acassis and @gustavonihei please take a look.

**Review by gustavonihei — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
It pass CI except "Mix case" warning, @gustavonihei please take a look.

**Review by gustavonihei — APPROVED** [review]

**Pull request merged**